addons/l10n_tw_edi_ecpay_website_sale/static/src/js/payment_form.js

```diff
@@ -0,0 +1,303 @@
+        }
+        this.validCarrierNumber = false;
+        this.el.querySelector("#warning-l10n_tw_edi_carrier_number").classList.remove("d-none");
+        this.showInvoiceItems();
+    },
+
+    _onChangeIdentifierGroup(ev) {
+        this.showIdentifierData = Boolean(parseInt(ev.target.value, 10));
+        this.showInvoiceItems();
+    },
+
+    _onInputIdentifier: function (ev) {
+        this.validIdentifier = false;
+    },
+
+    _onInputCarrierNumber(ev) {
+        const carrierType = this.el.querySelector("#l10n_tw_edi_carrier_type").value;
+        if (carrierType === "2") {
+            this.el.querySelector("#warning-l10n_tw_edi_carrier_number").innerHTML =
+                "Correct Format: 2 capital letters following 14 digits";
+            const re = /^[A-Za-z]{2}[0-9]{14}$/;
+            this.validCarrierNumber = Boolean(re.test(ev.target.value));
+            this.triggerWarning(
+                "#warning-l10n_tw_edi_carrier_number",
+                "#ecpay_invoice_carrier_number",
+                this.validCarrierNumber
+            );
+        } else if (carrierType === "3") {
+            this.el.querySelector("#warning-l10n_tw_edi_carrier_number").innerHTML =
+                "Correct Format: '/' following 7 alphanumeric or +-. string";
+            const re = /^\/{1}[0-9a-zA-Z+-.]{7}$/;
+            if (re.test(ev.target.value)) {
+                this.el.querySelector("#validate_carrier_number").classList.remove("d-none");
+                this.triggerWarning("#warning-l10n_tw_edi_carrier_number", "#ecpay_invoice_carrier_number", true);
+            } else {
+                this.el.querySelector("#validate_carrier_number").classList.add("d-none");
+                this.triggerWarning(
+                    "#warning-l10n_tw_edi_carrier_number",
+                    "#ecpay_invoice_carrier_number",
+                    this.validCarrierNumber
+                );
+            }
+        } else {
+            this.el.querySelector("#validate_carrier_number").classList.add("d-none");
+            this.triggerWarning(
+                "#warning-l10n_tw_edi_carrier_number",
+                "#ecpay_invoice_carrier_number",
+                this.validCarrierNumber
+            );
+        }
+    },
+
+    _onInputLoveCode(ev) {
+        this.validLoveCode = false;
+        this.el.querySelector("#warning-l10n_tw_edi_love_code").innerHTML = "Love code format is 3-7 digits";
+        const re = /^([xX]{1}[0-9]{2,6}|[0-9]{3,7})$/;
+        if (re.test(ev.target.value)) {
+            this.el.querySelector("#validate_love_code").classList.remove("d-none");
+            this.triggerWarning("#warning-l10n_tw_edi_love_code", "#ecpay_invoice_love_code", true);
+        } else {
+            this.el.querySelector("#validate_love_code").classList.add("d-none");
+            this.triggerWarning("#warning-l10n_tw_edi_love_code", "#ecpay_invoice_love_code", this.validCarrierNumber);
+        }
+    },
```

Comment:
Adding a warning to give the expected format isn't the best. Use a placeholder for that.
Only add a warning when there is an error.